### PR TITLE
e2e: create and patch items of type Secret, code cleanup

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

This is useful for CSI drivers like ceph-csi which have to be
configured via secrets. I am now using the refactored e2e/framework out-of-tree and found that the extension mechanism that was meant to allow out-of-tree users of the framework to extend the item loading isn't sufficient at the moment because it doesn't cover patching. I am going to enhance that for 1.14 (issue #70668) but for 1.13 it would be good to get this merged.

**Special notes for your reviewer**:

I've tested this out-of-tree.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig storage

### CODE REVIEW

File: test/e2e/framework/create.go [hunk 1]
```diff
@@ -655,3 +658,32 @@ func PrettyPrint(item interface{}) string {
 	}
 	return fmt.Sprintf("%+v", item)
 }
+
+type SecretFactory struct{}
```
Comment (neolit123) [inline test/e2e/framework/create.go]: the type and methods would need export comments.

**Review by neolit123 — COMMENTED** [review]
/priority important-longterm

File: test/e2e/framework/create.go [hunk 2]
```diff
@@ -655,3 +658,32 @@ func PrettyPrint(item interface{}) string {
...
+	client := f.ClientSet.CoreV1().Secrets(f.Namespace.GetName())
+	if _, err := client.Create(item); err != nil {
+		return nil, errors.Wrap(err, "create Secret")
+	}
+	return func() error {
+		return client.Delete(item.GetName(), &metav1.DeleteOptions{})
+	}, nil
+}
+
+func (*SecretFactory) UniqueName(i interface{}) string {
```
Comment (mikedanese) [inline test/e2e/framework/create.go]: MetaNamespaceKeyFunc?

https://godoc.org/k8s.io/client-go/tools/cache#MetaNamespaceKeyFunc

File: test/e2e/framework/create.go (hunk 1 above)
Comment (mikedanese) [reply test/e2e/framework/create.go]: or you can just make it private.

File: test/e2e/framework/create.go (hunk 2 above)
Comment (pohly) [reply test/e2e/framework/create.go]: Thanks for the pointer. MetaNamespaceKeyFunc is not quite the same (UniqueName will ignore the namespace of items which are known to be not namespaced), but as this shouldn't matter in practice (namespace shouldn't be set then), I'll remove all of the UniqueName methods with it.

File: test/e2e/framework/create.go (hunk 1 above)
Comment (pohly) [reply test/e2e/framework/create.go]: Making them private sounds much better than adding mostly redundant comments just for the sake of it. Will do.

**Comment by pohly** [comment]
@neolit123 @mikedanese thanks for the prompt review. I've update the PR, please take another look.

**Review by neolit123 — COMMENTED** [review]
@pohly ideally the lowercase reactor could have been a separate PR / commit at least, from the addition of secrets, but the change is not that big, so this feels OK.

moving to private functions LGTM.

**Pull request merged**